tests/test_cephes.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import division
+import random
+from unittest import TestCase
+
+import numpy as np
+import scipy.special as special
+
+from cepy import bdtr
+from cepy import betaln
+from cepy import btdtr
+from cepy import gamln
+from cepy import gammainc
+from cepy import ndtri
+from cepy import pdtr
+
+
+np.seterr(all="ignore")
+
+
+def allClose(a, b):
+    global_truth = True
+    for i in range(len(a)):
+        local_truth = True
+        if round(a[i] - b[i], 5) != 0:
+            local_truth = False
+        if np.isnan(a[i]) and np.isnan(b[i]):
+            local_truth = True
+        if np.isinf(a[i]) and np.isinf(b[i]):
+            local_truth = True
+        global_truth = local_truth
+    return global_truth
+
+
+class _TestOneArgFn(object):
```

Comment:
There is a simpler way of having the same test with different arity functions: just have the test case specify the arity of the function in the `setUp()` (like `self.arity = 2`), and then do

```python
actual = starmap(self.fn, product(self.vals, repeat=self.arity))
expected = starmap(self.refFn, product(self.vals, repeat=self.arity))
```
Then you only need a single base mixin, like this:

```python
class ComparisonTestMixin(object):

    def assertAllClose(self, iterable1, iterable2):
        self.assertTrue(numpy.allclose(actual, expected))

    def test_fn(self):
        actual = starmap(self.fn, product(self.vals, repeat=self.arity))
        expected = starmap(self.refFn, product(self.vals, repeat=self.arity))
        self.assertAllClose(actual, expected)

class TestGamln(TestCase, ComparisonTestMixin):

    def setUp(self):
        self.arity = 1
        self.fn = gamln
        self.refFn = special.gammaln
        self.vals = [...]
```